### ISSUE TITLE
Wrong stride calculation in enc

### ISSUE THREAD

**Issue by lvqier** [JeffyCN/libv4l-rkmpp#28]
I am using h264 encoding, my input picture format is V4L2_PIX_FMT_UYVY, with resolution 1280x720. The right value of hstride should be 2560, but in rkmpp_enc_apply_input_cfg, it is 1280, which is calculated at line libv4l-rkmpp-enc.c#L570. The result image of video is shown as below:
![image](https://github.com/user-attachments/assets/c6b3c943-568d-49a4-b8aa-48ff6d2cd686)

After modifying the hstride calculation as below, h264 enc works properly.

`diff --git a/src/libv4l-rkmpp-enc.c b/src/libv4l-rkmpp-enc.c
index 0e43ea6..3ef0d81 100644
--- a/src/libv4l-rkmpp-enc.c
+++ b/src/libv4l-rkmpp-enc.c
@@ -36,6 +36,8 @@ enum v4l2_mpeg_video_bitrate_mode {

 #define RKMPP_ENC_POLL_TIMEOUT_MS	100

+#define MPP_ALIGN(x, a) (((x) + (a)-1) & ~((a)-1))
+
 static struct rkmpp_fmt rkmpp_enc_fmts[] = {
 	{
 		.name = "4:2:0 3 plane Y/Cb/Cr",
@@ -568,6 +570,7 @@ static int rkmpp_enc_apply_input_cfg(struct rkmpp_enc_context *enc)
 	enc->width = enc->crop.width ? enc->crop.width : fmt->width;
 	enc->height = enc->crop.height ? enc->crop.height : fmt->height;
 	enc->hstride = fmt->plane_fmt[0].bytesperline * 8 / rkmpp_fmt->depth[0];
+        enc->hstride = MPP_ALIGN(fmt->width * 2, 16);
 	enc->vstride = fmt->plane_fmt[0].sizeimage /
 		fmt->plane_fmt[0].bytesperline;
`

**Comment by JeffyCN** [comment]
i've uploaded a patch for this:)

**Comment by lvqier** [comment]
Incorrect, hstride turns into 10240 in my case. Encoded picture is shown:
![image](https://github.com/user-attachments/assets/d7471fb6-cdc2-485b-b8c6-8224af91e308)

**Comment by JeffyCN** [comment]
re-uploaded

**Comment by lvqier** [comment]
Working now. Thanks!
